source/Halibut/Transport/ConnectionManager.cs

```diff
@@ -143,6 +143,10 @@ public void Dispose()
             IsDisposed = true;
         }
 
+        public ValueTask DisposeAsync()
+        {
+            throw new NotImplementedException("Should not be called when async Halibut is not being used.");
```

Comment:
Nit: I haven't got a strong preference, but we have 2 different ways of saying don't call this cuz your sync/async via `NotSupportedException` and then `NotImplementedException` types.

Should we keep the exception type consistent?

![image](https://github.com/OctopusDeploy/Halibut/assets/86938706/ac2d9a10-d140-44aa-9827-5983f466a183)

Reply:
We totally should! Done 🙂